projects/packages/jetpack-mu-wpcom/src/features/wpcom-command-palette/wpcom-command-palette.php

```diff
@@ -36,7 +36,18 @@ function wpcom_load_command_palette() {
 	wp_enqueue_script(
 		$command_palette_js_handle,
 		'//widgets.wp.com/command-palette/build.min.js',
-		array(),
+		array(
+			'react',
+			'react-dom',
+			'wp-components',
+			'wp-compose',
+			'wp-dom-ready',
+			'wp-element',
+			'wp-i18n',
+			'wp-polyfill',
+			'wp-primitives',
+			'wp-url',
+		),
```

Comment:
Ideally, we should fetch these dependencies from the generated `build.asset.php` file, but Atomic sites won't have access to that file unless it is fetched remotely from widgets.wp.com/command-palette/build.asset.php.

The file is rather small, but I'm concerned about making that request on every single request to WP Admin, so I thought a hardcoded list would be fine for enough. We just need to remember this if we add a new dependency to the command palette.

Reply:
Probably ought to be noted somewhere more permanently like the command palette app readme perhaps

Reply:
https://github.com/Automattic/wp-calypso/pull/88193/commits/d2b9e54d1fd6ccc59f8fdd6bfc178bafcc50ae68

Reply:
Yeah the indirection here isn't ideal; jetpack, calypso, widgets... You might've considered already but wanted to list some options in case it helps:

- Pull the command palette package developed in calypso as a package to Jetpack, and require the PHP file that way.
- Move all command palette code JS to Jetpack alongside PHP code, and pull the package into Calypso. Otherwise wp-admin and Calypso just load Jetpack regularly.

In my experience keeping PHP and JS in separately across wp-calypso/jetpack or even utilizing widgets to ship to Atomic makes for less ideal devex.

Reply:
😆 not taking you up on that bet!

Ultimately we should use the GB command palette and then some of these problems go away.

Reply:
Yeah, I recognize that having local access to the build in both Jetpack and Calypso would be ideal, but it didn't seem trivial to achieve that when we first looked into it.

> Pull the command palette package developed in Calypso as a package to Jetpack, and require the PHP file that way.

- Calypso public packages available in npm cannot contain PHP files, so Jetpack cannot use a public package to require a PHP file.
- Calypso private apps can contain PHP files, but they are not publicly available, so Jetpack cannot use them.

A possible workaround would be to have something similar to the old Fusion system that automatically creates a counterpart PR in Jetpack to have Calypso and Jetpack in sync, but I'm not sure whether this effort would contribute towards the ideal devex.

> Move all command palette code JS to Jetpack alongside PHP code, and pull the package into Calypso. Otherwise, wp-admin and Calypso just load Jetpack regularly.

This sounds plausible, but I admit that I'm not familiar at all with how JS packages are published to npm and if they can be consumed by Calypso (a first look reveals that Calypso does not use any `@automattic/jetpack-*` package).

Said that, ultimately we want to use the Gutenberg command palette (`@wordpress/commands`) and once we can do that, I think we can get rid of most of the indirection here.

Reply:
> Said that, ultimately we want to use the Gutenberg command palette (@wordpress/commands) and once we can do that, I think we can get rid of most of the indirection here.

It's still a bit hassle for commands that could be shared across Jetpack plugin, WP.com wp-admin and Calypso (noted at p9Jlb4-b8X-p2/#comment-10959 ).